src/table/components/TableHeadWrapper.jsx

```diff
@@ -21,13 +22,19 @@ function TableHeadWrapper({
   children,
 }) {
   const { columns, paginationNeeded } = tableData;
+  const setHeadRowHeight = useContextSelector(TableContext, (value) => value.setHeadRowHeight);
   const isFocusInHead = useContextSelector(TableContext, (value) => value.focusedCellCoord[0] === 0);
   const setFocusedCellCoord = useContextSelector(TableContext, (value) => value.setFocusedCellCoord);
   const headerStyle = useMemo(() => getHeaderStyle(layout, theme), [layout, theme]);
+  const headRowRef = useRef();
+
+  useEffect(() => {
+    setHeadRowHeight(headRowRef.current.clientHeight);
+  }, [headRowRef.current, headerStyle.fontSize]);
```

Comment:
removing `headRowRef.current` wont make it listen to `headerStyle.fontSize`, right now it will listen to both of them. the way the code looks right now: when `layout` or `theme` changes, the `headerStyle` will be updated. If the `fontSize` has changed, `setHeadRowHeight` will run. Now if `headRowRef.current.clientHeight` is updated to the new height, this should work